src/components/enterprise-customer-redirect/data/hooks.js

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState, useMemo } from 'react';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+import { logError } from '@edx/frontend-platform/logging';
+
+import { fetchEnterpriseCustomersForUser } from './service';
+
+/**
+ * A React hook that will fetch all the enterprise customers linked to the specified user id.
+ *
+ * @param {*} userId A user's id.
+ *
+ * @returns {list} obj.enterpriseCustomers A list of enterprise customers associated with a user. Note this is empty
+ *  until the fetch promise has finished resolving.
+ * @returns {bool} obj.isLoading A Boolean for whether the request is pending.
+ */
+export const useEnterpriseCustomersForUser = (userId) => {
+  const [enterpriseCustomers, setEnterpriseCustomers] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(
+    () => {
+      if (!userId) {
+        return;
+      }
+
+      fetchEnterpriseCustomersForUser()
```

Comment:
`userId` is not used in this call?

Reply:
The `enterprise-customers` API only returns enterprises for which the user is linked. E.g., if I'm only linked to 1 customer, it will only return that 1 result.

If I'm not linked to any customers, it will return an empty list.

The endpoint handles this user permissions check via edx-rbac.